src/linked_list/linked_list.ts

```diff
@@ -0,0 +1,139 @@
+interface INode<T> {
+  data: T;
+  next: INode<T> | null;
+}
+
+export class Node<T> implements INode<T> {
+  data: T;
+  next: INode<T> | null;
+
+  constructor(data: T) {
+    this.data = data;
+    this.next = null;
+  }
+}
+
+interface ILinkedList<T> {
+  addFirst: (data: T) => void;
+  addLast: (data: T) => void;
+  getFirst: () => T | null;
+  getLast: () => T | null;
+  removeFirst: () => T | null;
+  removeLast: () => T | null;
+  size: () => number;
+  clear: () => void;
+  toArray: () => T[];
+}
+
+export class LinkedList<T> implements ILinkedList<T> {
+  private head: INode<T> | null;
+
+  constructor() {
+    this.head = null;
+  }
+
+  addFirst(data: T) {
+    const newNode = new Node(data);
+
+    newNode.next = this.head;
+    this.head = newNode;
+  }
+
+  addLast(data: T) {
+    const newNode = new Node(data);
+
+    if (!this.head) {
+      this.head = newNode;
+
+      return;
+    }
+
+    let currentNode = this.head;
+
+    while (currentNode.next) {
+      currentNode = currentNode.next;
+    }
+
+    currentNode.next = newNode;
+  }
+
+  getFirst() {
+    if (!this.head) {
+      return null;
+    }
+
+    return this.head.data
+  }
+
+  getLast() {
+    if (!this.head) {
+      return null;
+    }
+
+    let currentNode = this.head;
+
+    while (currentNode.next) {
+      currentNode = currentNode.next;
+    }
+
+    return currentNode.data;
+  }
+
+  removeFirst() {
+    if (!this.head) {
+      return null;
+    }
+
+    const data = this.head.data;
+
```

Comment:
Here the empty line looks 100% unnecessary. Try not to add many blank lines, only where it can add extra readability